app/models/age_requirement.rb

```diff
@@ -0,0 +1,27 @@
+class AgeRequirement
+  include Neo4j::ActiveNode
+  has_many :out, :requirements, type: :HAS_REQUIREMENT, model_class: [:OrRequirementGroup, :AndRequirementGroup]
+  property :min_age
+  property :max_age
+
+  def ineligible?(attributes)
+    return false unless attributes[:age].present?
+
+    met_requirements = requirements_present.map { |requirement| send("meets_#{requirement}_requirement?", attributes) }
+    !met_requirements.all?
+  end
+
+  private
+
+  def requirements_present
+    %i(min_age max_age).select { |requirement| read_attribute(requirement).present? }
+  end
```

Comment:
As a suggestion, could we eliminate the string interpolation `ineligible?` by doing something like:

```
criteria = {min_age: "meets_min_age_requirement?", max_age: "meets_max_age_requirement?}
%(min_age max_age).select do |requirement|
  criteria[requirement] if read_attribute(requirement).present?
end
```

then in `ineligible?` we could just `send(requirement)`? 

Haven't tested it, so I'm not sure if that would work.

Reply:
This is a much better idea, changed